src/routes/(console)/project-[region]-[project]/functions/create-function/deploy/+page.svelte

```diff
@@ -0,0 +1,345 @@
+        Layout,
+        Icon,
+        Typography,
+        Input,
+        Tag,
+        Selector
+    } from '@appwrite.io/pink-svelte';
+    import { IconGithub, IconExternalLink, IconPencil } from '@appwrite.io/pink-icons-svelte';
+    import { writable } from 'svelte/store';
+    import { onMount } from 'svelte';
+    import { ID, Runtime } from '@appwrite.io/console';
+    import { CustomId } from '$lib/components';
+    import { getIconFromRuntime } from '$lib/stores/runtimes';
+    import { regionalConsoleVariables } from '$routes/(console)/project-[region]-[project]/store';
+    import { iconPath } from '$lib/stores/app';
+    import type { PageData } from './$types';
+
+    export let data: PageData;
+
+    let showExitModal = false;
+    let showCustomId = false;
+    let isSubmitting = writable(false);
+
+    // State variables
+    let name = '';
+    let id = '';
+    let runtime: Runtime;
+    let entrypoint = '';
+    let buildCommand = ''; // This is the install command in the UI
+    let rootDir = './';
+    let variables: Array<{ key: string; value: string; secret: boolean }> = [];
+    let specification = '';
+    let execute = true;
+    let selectedScopes: string[] = [];
+
+    // Track the latest tag (needed for deployment, but not shown in UI)
+    let latestTag = '';
+
+    const specificationOptions =
+        data.specificationsList?.specifications?.map((size) => ({
+            label:
+                `${size.cpus} CPU, ${size.memory} MB RAM` +
+                (!size.enabled ? ` (Upgrade to use this)` : ''),
+            value: size.slug,
+            disabled: !size.enabled
+        })) || [];
+
+    onMount(() => {
+        // Initialize default values
+        name = data.repository.name;
+        id = ID.unique();
+
+        // Get URL params
+        const runtimeParam = data.runtime || page.url.searchParams.get('runtime') || 'node-18.0';
+        runtime = runtimeParam as Runtime;
+
+        // Build configuration - use from URL params or defaults
+        entrypoint = page.url.searchParams.get('entrypoint') || '';
+        buildCommand = page.url.searchParams.get('install') || ''; // Using 'install' param for consistency with sites
+        rootDir = page.url.searchParams.get('rootDir') || './';
+
+        // Set default specification
+        if (specificationOptions.length > 0) {
+            specification = specificationOptions[0].value;
+        }
+
+        // Initialize environment variables from query params
+        if (data.envKeys.length > 0) {
+            variables = data.envKeys.map((key) => ({ key, value: '', secret: false }));
+        }
+
+        // Fetch latest tag in the background (needed for deployment)
+        fetchLatestGitHubTag();
+    });
+
+    async function fetchLatestGitHubTag(): Promise<void> {
+        try {
+            const tagsResponse = await fetch(
+                `https://api.github.com/repos/${data.repository.owner}/${data.repository.name}/tags`
+            );
```

Comment:
Template response has `providerVersion`, which is for example `0.2.*`.
We should search for such version, so for example, here we would get `0.2.4`, because that's the latest version that matches version wildcard `0.2.*`